packages/drivers/odsp-driver/src/compactSnapshotParser.ts

```diff
@@ -0,0 +1,156 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { assert } from "@fluidframework/common-utils";
+import { ISnapshotTree } from "@fluidframework/protocol-definitions";
+import { ISequencedDeltaOpMessage } from "./contracts";
+import { BlobCore, NodeCore, ReadBuffer, TreeBuilder } from "./zipItDataRepresentationUtils";
+
+export const snapshotMinReadVersion = "1.0";
+
+/**
+ *  Header of the downloaded snapshot.
+ */
+export interface ISnapshotHeader {
+    // This is the minimum version of the reader required to read the wire format of a given snapshot.
+    minReadVersion: string,
+    // Represents the version with which snapshot is created.
+    createVersion: string,
+    // Seq number at which the snapshot is created.
+    snapshotSeqNumber: number,
+}
+
+/**
+ * Recreates header section and then validate that.
+ * @param node - tree node to read header section from
+ */
+function readAndValidateHeaderSection(node: NodeCore) {
+    let minReadVersion: string | undefined;
+    let createVersion: string | undefined;
+    for (const [headerName, value] of node.iteratePairs()) {
```

Comment:
Consider if we can make it more generic / reusable, to be useful in other cases where we might have property bags.
For example, if we assume that every property would be of string type, then I'd rather have generic function that takes a node and returns JS object with property bag. Maybe have another function that takes an object and validates that it has certain required properties (passed as an array).

Reply:
Added an api that will parse and check that a node has certain required properties. I think the other suggestion about adding  some generic function that returns JS object of generic props may not be of that much importance as then we would be iterating over props twice as we don't know what they would be. However it makes sense for known props and I added an api for that.